---
title: "Clustering of All Gut Samples using Taxonomic Abundance"
author: "Levi Waldron"
date: "`r format(Sys.time(), '%B %d, %Y')`"
output:
  html_document:
    number_sections: yes
    toc: yes
vignette: >
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteIndexEntry{Fetch and Cluster Gut Metagenomic Taxonomic Tables}
---

# Create datasets

Create an ExperimentHub containing metadata for all `curatedMetagenomicData` records:

```{r}
suppressPackageStartupMessages(library(ExperimentHub))
eh = ExperimentHub()
myquery = query(eh, "curatedMetagenomicData")
```

Subset this to bug abundance from stool datasets. Note that this could also be done using `myquery$tags` (once the tags are updated). See the available metadata using `View(mcols(myquery))`.

```{r}
myquery.stool <- myquery[grepl("stool", myquery$title) & grepl("bugs", myquery$title), ]
myquery.stool
```

Create a list of ExpressionSet objects:
```{r, message=FALSE}
eset.list <- lapply(names(myquery.stool), function(x) myquery.stool[[x]])
```

Give them simplified titles:
```{r}
names(eset.list) <- myquery.stool$title
names(eset.list) <- gsub("\\..+", "", myquery.stool$title)
```

And add the titles to the colnames:
```{r}
for (i in 1:length(eset.list)){
    colnames(eset.list[[i]]) <- paste(names(eset.list)[[i]],
                                      colnames(eset.list[[i]]), 
                                      sep=".")
    pData(eset.list[[i]]) <- pData(eset.list[[i]])[, !sapply(pData(eset.list[[i]]), function(x) all(is.na(x)))]
    eset.list[[i]]$subjectID <- as.character(eset.list[[i]]$subjectID)
}
```

# Merging into one big `ExpressionSet`

```{r, message=FALSE}
joinWithRnames <- function(obj, FUN = I) {
    mylist <- lapply(obj, function(x) {
        df <- data.frame(FUN(x))
        df$rnames28591436107 <- rownames(df)
        return(df)
    })
    bigdf <- Reduce(dplyr::full_join, mylist)
    rownames(bigdf) <- make.names(bigdf$rnames28591436107)
    bigdf <- bigdf[, !grepl("^rnames28591436107$", colnames(bigdf))]
    return(bigdf)
}

pdat <- joinWithRnames(eset.list, FUN=pData)
pdat$study <- sub("\\..+", "", rownames(pdat))
ab <- joinWithRnames(eset.list, FUN=exprs)
eset <- ExpressionSet(assayData = as.matrix(ab), phenoData = AnnotatedDataFrame(pdat))
eset
```

# Convert to `phyloseq` object

```{r}
suppressPackageStartupMessages(library(phyloseq))
source("https://raw.githubusercontent.com/waldronlab/presentations/master/Waldron_2016-06-07_EPIC/metaphlanToPhyloseq.R")
pseq <- metaphlanToPhyloseq(tax=exprs(eset), metadat=pData(eset))
```

# Bray-Curtis distance ordination plot

```{r, cache=TRUE}
ord = ordinate(pseq, method="PCoA", distance="bray")
```

```{r}
plot_ordination(pseq, ord, color="disease") + 
  ggplot2::ggtitle("Bray-Curtis Principal Coordinates Analysis")
plot_ordination(pseq, ord, color="country") + 
  ggplot2::ggtitle("Bray-Curtis Principal Coordinates Analysis")
plot_ordination(pseq, ord, color="study") + 
  ggplot2::ggtitle("Bray-Curtis Principal Coordinates Analysis")
```

# Ugly summary statistics

```{r}
sapply(pdat, function(x){
    if(is(x, "numeric")){
        print(summary(x))
    }else{
      print(table(x))
    }}
)
```

# Session Info

```{r}
sessionInfo()
```

